ddtrace/contrib/gevent/__init__.py

```diff
@@ -1,62 +1,29 @@
 """
-To trace a request in a gevent-ed environment, configure the tracer to use greenlet-local
-storage, rather than the default thread-local storage.
+To trace a request in a gevent-ed environment, configure the tracer to use the Greenlet
+context provider, rather than the default one that relies in thread-local storaging.
 
-This allows the tracer to pick up a transaction exactly
-where it left off as greenlets yield context to one another.
+This allows the tracer to pick up a transaction exactly where it left off as greenlets
+yield the context to one another.
 
-The simplest way to trace with greenlet-local storage is via the `gevent.monkey` module::
+The simplest way to trace a ``gevent`` application is to configure the tracer and
+patch gevent before using it::
 
-    # Always monkey patch before importing the global tracer
-    # Broadly, gevent recommends that patches happen as early as possible in the app lifecycle
-    # http://www.gevent.org/gevent.monkey.html#patching
+    # Always monkey patch before importing gevent
+    from ddtrace import patch, tracer
+    from ddtrace.contrib.gevent import context_provider
 
-    from gevent import monkey; monkey.patch_thread()
-    # Alternatively, use monkey.patch_all() to perform all available patches
-
-    from ddtrace import tracer
-
-    import gevent
-
-    def my_parent_function():
-        with tracer.trace("web.request") as span:
-            span.service = "web"
-            gevent.spawn(worker_function, span)
-
-    def worker_function(parent):
-        # Set the active span
-        tracer.span_buffer.set(parent)
-
-        # then trace its child
-        with tracer.trace("greenlet.call") as span:
-            span.service = "greenlet"
-            ...
+    tracer.configure(context_provider=context_provider)
+    patch(gevent=True)
```

Comment:
actually we don't have a way to patch things using the `configure()` method. I don't like having these two lines too because they're error prone (both are required). What I can do is:
* having the class for this context provider that patches automatically the code like:
```python
# even if not strictly required, it's better that the
# patch happens before importing gevent
from ddtrace import tracer
from ddtrace.contrib.gevent import GeventContextProvider
tracer.configure(context_provider=GeventContextProvider())

import gevent
# do whatever you need
```
In that case, under the hood, the constructor will call the gevent `patch()`